File: prism/backend/src/routes/analyzeRoute.ts
/**
 * =============================================================================
 * FILE PURPOSE
 * =============================================================================
 *
 * This file defines the HTTP route for content analysis: accept POST body
 * (URL, title, extracted text), call backend services to run LLM analysis
 * and bias detection, then return a response that conforms to the shared
 * analysis schema.
 *
 * =============================================================================
 * OWNER ROLE
 * =============================================================================
 *
 * Backend Lead
 *
 * =============================================================================
 * RESPONSIBILITIES
 * =============================================================================
 *
 * - Define POST handler for the analyze endpoint.
 * - Validate incoming body (required fields, basic types) and reject bad
 *   requests with 400 and a clear message.
 * - Sanitize/truncate input using utils (textSanitizer, truncation) before
 *   passing to services.
 * - Invoke llmService (with prompt from promptBuilder), then run
 *   responseValidator on the raw LLM output. Optionally run biasDetection
 *   logic and merge into the response.
 * - Return 200 with schema-shaped JSON; on LLM or validation failure,
 *   return 500 or 422 with error details.
 * - No prompt or model logic here; delegate to services and logic.
 *
 * =============================================================================
 * INTEGRATION NOTES
 * =============================================================================
 *
 * - Mounted by server.ts (e.g. app.use('/api/analyze', analyzeRoute)).
 * - Uses services/llmService.ts, services/promptBuilder.ts,
 *   services/responseValidator.ts and logic/biasDetection.ts.
 * - Uses utils/textSanitizer.ts and utils/truncation.ts for input safety.
 * - Request/response shapes must align with shared/schema/analysisSchema.json.
 * - Extension api/analyzeClient.ts calls this endpoint.
 *
 * =============================================================================
 */

File: prism/backend/src/routes/analyzeRoute.py
# =============================================================================
# FILE PURPOSE
# =============================================================================
#
# This file defines the HTTP route for content analysis: accept POST body
# (URL, title, extracted text), call backend services to run LLM analysis
# and bias detection, then return a response that conforms to the shared
# analysis schema.
#
# =============================================================================
# OWNER ROLE
# =============================================================================
#
# Backend Lead
#
# =============================================================================
# RESPONSIBILITIES
# =============================================================================
#
# - Define POST handler for the analyze endpoint.
# - Validate incoming body (required fields, basic types) and reject bad
#   requests with 400 and a clear message.
# - Sanitize/truncate input using utils (textSanitizer, truncation) before
#   passing to services.
# - Invoke llmService (with prompt from promptBuilder), then run
#   responseValidator on the raw LLM output. Optionally run biasDetection
#   logic and merge into the response.
# - Return 200 with schema-shaped JSON; on LLM or validation failure,
#   return 500 or 422 with error details.
# - No prompt or model logic here; delegate to services and logic.
#
# =============================================================================
# INTEGRATION NOTES
# =============================================================================
#
# - Mounted by server.py (e.g. app.include_router(analyze_route)).
# - Uses services/llm_service.py, services/prompt_builder.py,
#   services/response_validator.py and logic/bias_detection.py.
# - Uses utils/text_sanitizer.py and utils/truncation.py for input safety.
# - Request/response shapes must align with shared/schema/analysisSchema.json.
# - Extension api/analyzeClient.ts calls this endpoint.
#
# =============================================================================


from fastapi import APIRouter, HTTPException

from src.utils.textSanitizer import sanitize_text
from src.utils.truncation import truncate_text

router = APIRouter(prefix="/api/analyze", tags=["analyze"])


@router.post("/")
async def analyze_content(payload: dict):

    # validation
    required_fields = ["url", "title", "text"]
    for field in required_fields:
        if field not in payload:
            raise HTTPException(status_code=400, detail="Missing text field")

    # sanitize
    clean_title = sanitize_text(payload["title"])
    clean_text = sanitize_text(payload["text"])

    # truncate
    final_text = truncate_text(clean_text, max_chars=6000)

    # possible for when everythin else is built out

    ## build prompt
    # prompt = build_prompt(
    #     url=payload["url"],
    #     title=clean_title,
    #     text=final_text
    # )
    #
    # # call LLM
    # raw_output = await call_llm(prompt)
    #
    # # validate response schema
    # structured_output = validate_response(raw_output)
    #
    # # optional bias detection layer
    # bias_result = detect_bias(final_text)
    #
    # # merge final response
    # structured_output["bias"] = bias_result

    # call services here later
    return {"status": "success", "analysis": {}}
